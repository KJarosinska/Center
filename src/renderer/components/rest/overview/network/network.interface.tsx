import SimInfo from "Common/interfaces/sim-info.interface"

export interface NetworkProps {
  simCards?: SimInfo[]
}
