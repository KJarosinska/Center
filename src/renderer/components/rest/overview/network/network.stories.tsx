import { storiesOf } from "@storybook/react"
import React from "react"
import Network from "Renderer/components/rest/overview/network/network.component"
import getFakeAdapters from "App/tests/get-fake-adapters"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import { action } from "@storybook/addon-actions"

storiesOf("Overview/Network", module).add("Basic", () => {
  return (
    <div style={{ margin: "2rem", maxWidth: "59rem" }}>
      <Text displayStyle={TextDisplayStyle.SmallText}>No SIM card</Text>
      <br />
      <Network />
      <br />
      <br />
      <Text displayStyle={TextDisplayStyle.SmallText}>
        One SIM card (active)
      </Text>
      <br />
      <Network
        simCards={getFakeAdapters()
          .pureNetwork.getSimCards()
          .slice(0, 1)}
      />
      <br />
      <br />
      <Text displayStyle={TextDisplayStyle.SmallText}>
        One SIM card (inactive)
      </Text>
      <br />
      <Network
        simCards={getFakeAdapters()
          .pureNetwork.getSimCards()
          .slice(1, 2)}
        onSimChange={action("SIM changed")}
      />
      <br />
      <br />
      <Text displayStyle={TextDisplayStyle.SmallText}>Two SIM cards</Text>
      <br />
      <Network
        simCards={getFakeAdapters().pureNetwork.getSimCards()}
        onSimChange={action("SIM changed")}
      />
    </div>
  )
})
