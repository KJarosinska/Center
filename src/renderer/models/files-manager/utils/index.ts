import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
