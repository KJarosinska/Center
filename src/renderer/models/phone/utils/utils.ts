import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
