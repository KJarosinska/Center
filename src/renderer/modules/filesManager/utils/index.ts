import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
