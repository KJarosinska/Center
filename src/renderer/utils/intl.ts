import { createIntl } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"

export const intl = createIntl({
  locale: LANGUAGE.default,
  messages: localeEn,
})
