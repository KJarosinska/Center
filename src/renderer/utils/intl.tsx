import React from "react"
import { createIntl } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"

const testLocale = extractLanguageKeys(localeEn)

export const intl = createIntl({
  locale: LANGUAGE.default,
  messages: process.env.NODE_ENV === "test" ? testLocale : localeEn,
})

export const textFormatters = {
  b: (str: string) => <strong>{str}</strong>,
  u: (str: string) => <u>{str}</u>,
  i: (str: string) => <em>{str}</em>,
  s: (str: string) => <s>{str}</s>,
}
