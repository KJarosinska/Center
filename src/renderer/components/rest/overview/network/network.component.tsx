import React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import { NetworkProps } from "Renderer/components/rest/overview/network/network.interface"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import { FormattedMessage } from "react-intl"
import styled from "styled-components"
import { intl } from "Renderer/utils/intl"
import SimInfo from "Common/interfaces/sim-info.interface"
import { textColor } from "Renderer/styles/theming/theme-getters"
import Card, {
  CardAction,
  CardActionButton,
  CardText,
} from "Renderer/components/rest/overview/card.elements"

const TextInfo = styled(CardText)`
  p {
    margin-top: 0.8rem;
    color: ${textColor("placeholder")};
  }
`

const SimButton = ({ slot, number: phone, active }: SimInfo) => {
  const label = intl.formatMessage(
    {
      id: "view.name.overview.network.simInfo",
    },
    { slot, phone }
  )
  return <CardActionButton label={label} active={active} />
}

const NoSimButton = () => {
  const label = intl.formatMessage({
    id: "view.name.overview.network.noSimInserted",
  })
  return <CardActionButton label={label} disabled />
}

const Network: FunctionComponent<NetworkProps> = ({
  className,
  simCards = [],
}) => {
  const noActiveCard = simCards.every(({ active }) => !active)
  const activatingAvailable =
    (simCards.length && noActiveCard) || simCards.length > 1
  return (
    <Card className={className}>
      <TextInfo>
        <Text displayStyle={TextDisplayStyle.SecondaryBoldHeading}>
          <FormattedMessage id="view.name.overview.network.name" />
        </Text>
        {activatingAvailable && (
          <Text displayStyle={TextDisplayStyle.MediumText}>
            <FormattedMessage id="view.name.overview.network.description" />
          </Text>
        )}
      </TextInfo>
      <CardAction>
        {Boolean(simCards.length) ? (
          simCards.map(simCard => (
            <SimButton key={simCard.number} {...simCard} />
          ))
        ) : (
          <NoSimButton />
        )}
      </CardAction>
    </Card>
  )
}

export default Network
