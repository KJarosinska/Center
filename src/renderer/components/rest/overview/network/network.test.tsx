import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import { NetworkProps } from "Renderer/components/rest/overview/network/network.interface"
import Network from "Renderer/components/rest/overview/network/network.component"
import { intl } from "Renderer/utils/intl"
import getFakeAdapters from "App/tests/get-fake-adapters"

const renderNetwork = ({ simCards = [] }: Partial<NetworkProps> = {}) => {
  const outcome = renderWithThemeAndIntl(<Network simCards={simCards} />)
  return {
    ...outcome,
    getButtons: () => outcome.queryAllByRole("button"),
    getDescription: () =>
      outcome.queryByText(
        intl.formatMessage({ id: "view.name.overview.network.description" })
      ),
  }
}

test("matches snapshot", () => {
  const { container } = renderNetwork()
  expect(container).toMatchSnapshot()
})

test("renders card name properly", () => {
  const { queryByText } = renderNetwork()
  expect(
    queryByText(intl.formatMessage({ id: "view.name.overview.network.name" }))
  )
})

test("renders 'no sim card' state properly", () => {
  const { getButtons, getDescription } = renderNetwork()
  expect(getButtons()).toHaveLength(1)
  expect(getButtons()[0]).toHaveTextContent(
    intl.formatMessage({ id: "view.name.overview.network.noSimInserted" })
  )
  expect(getDescription()).not.toBeInTheDocument()
})

test("renders single active sim card info properly", () => {
  const simCard = getFakeAdapters().pureNetwork.getSimCards()[0]
  const { getButtons, getDescription } = renderNetwork({
    simCards: [simCard],
  })
  expect(getButtons()).toHaveLength(1)
  expect(getButtons()[0]).toHaveTextContent(
    intl.formatMessage(
      {
        id: "view.name.overview.network.simInfo",
      },
      { slot: simCard.slot, phone: simCard.number }
    )
  )
  expect(getDescription()).not.toBeInTheDocument()
})

test("renders single inactive sim card info properly", () => {
  const simCard = getFakeAdapters().pureNetwork.getSimCards()[1]
  const { getButtons, getDescription } = renderNetwork({
    simCards: [simCard],
  })
  expect(getButtons()).toHaveLength(1)
  expect(getButtons()[0]).toHaveTextContent(
    intl.formatMessage(
      {
        id: "view.name.overview.network.simInfo",
      },
      { slot: simCard.slot, phone: simCard.number }
    )
  )
  expect(getDescription()).toBeInTheDocument()
})

test("renders two sim cards info properly", () => {
  const simCards = getFakeAdapters().pureNetwork.getSimCards()
  const { getButtons, getDescription } = renderNetwork({
    simCards,
  })
  expect(getButtons()).toHaveLength(2)
  simCards.forEach((simCard, index) => {
    expect(getButtons()[index]).toHaveTextContent(
      intl.formatMessage(
        {
          id: "view.name.overview.network.simInfo",
        },
        { slot: simCard.slot, phone: simCard.number }
      )
    )
  })
  expect(getDescription()).toBeInTheDocument()
})
