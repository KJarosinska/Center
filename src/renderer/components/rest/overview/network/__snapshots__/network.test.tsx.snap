// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c2 {
  margin: 0;
  font-size: 2.4rem;
  line-height: 0.8;
  font-weight: 700;
  -webkit-letter-spacing: -0.06rem;
  -moz-letter-spacing: -0.06rem;
  -ms-letter-spacing: -0.06rem;
  letter-spacing: -0.06rem;
}

.c9 {
  margin: 0;
  font-size: 1.4rem;
  line-height: 1.05;
}

.c11 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  color: #6a6a6a;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  font-weight: 300;
}

.c14 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 1.6rem;
  width: 1.6rem;
  position: relative;
}

.c5 svg {
  width: 100%;
  height: 100%;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  pointer-events: none;
  height: 4rem;
  border-radius: 0.5rem;
  border: 0.1rem solid #3b3f42;
  border: 0.1rem solid #cdcecf;
  color: #6a6a6a;
}

.c12 g {
  fill: #000000;
}

.c12 g {
  fill: #cdcecf;
}

.c12:hover {
  border-color: #000000;
}

.c15 {
  white-space: nowrap;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
}

.c8 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 24.3rem;
  background-color: #ffffff;
  padding: 1.6rem;
  position: absolute;
  top: 1rem;
  left: 0;
  opacity: 0;
  -webkit-transition: opacity 0.3s ease-in-out;
  transition: opacity 0.3s ease-in-out;
  box-shadow: 0 1rem 5.5rem -0.5rem rgba(214,214,214,0.5);
  z-index: 4;
}

.c6 {
  position: absolute;
  top: -0.8rem;
  left: -0.8rem;
  z-index: 4;
}

.c6:hover + .c7 {
  visibility: visible;
  opacity: 1;
}

.c10 {
  margin-bottom: 0.8rem;
}

.c0 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-areas: "Text Buttons";
  grid-template-columns: auto minmax(18rem,1fr);
  height: 14.4rem;
  padding: 0 4.8rem;
  box-sizing: border-box;
  border-radius: 0.5rem;
  background-color: #ffffff;
}

.c4 {
  grid-area: Buttons;
  justify-self: end;
  min-width: 18rem;
}

.c13 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  padding: 0 1.6rem;
  width: 50%;
}

.c13:disabled {
  pointer-events: none;
}

.c13:hover {
  z-index: 2;
}

.c13:not(:hover) {
  border: 0.1rem solid #cdcecf;
  color: #6a6a6a;
}

.c13:not(:hover) g {
  fill: #cdcecf;
}

.c13:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c13:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c13:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c13 svg {
  height: initial;
  width: initial;
}

.c1 {
  grid-area: Text;
}

.c1 p {
  margin-top: 1.2rem;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  color: #6a6a6a;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        [value] view.name.overview.network.name
      </h2>
    </div>
    <section
      class="c3 c4"
    >
      <div
        class="c5 c6"
        data-testid="icon-Tooltip"
        height="1.6"
        width="1.6"
      >
        <test-file-stub
          image="test-file-stub"
        />
      </div>
      <div
        class="c7 c8"
        data-testid="tooltip"
      >
        <p
          class="c9 c10"
        >
          [value] view.name.overview.network.tooltipTitle
        </p>
        <p
          class="c11"
        >
          [value] view.name.overview.network.tooltipDescription
        </p>
      </div>
      <button
        class="c12 c13"
        disabled=""
        size="2"
        type="button"
      >
        <p
          class="c14 c15"
        >
          [value] view.name.overview.network.noSimInserted
        </p>
      </button>
    </section>
  </section>
</div>
`;
